tests/integration/test_docs.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2022 the GPflow authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import subprocess
+from pathlib import Path
+
+import pytest
+
+BUILD_DOCS_PATH = Path(__file__).parent.parent.parent / "doc" / "build_docs.py"
+
+
+@pytest.mark.docs
+def test_docs(tmp_path: Path) -> None:
+    assert BUILD_DOCS_PATH.is_file()
+    build_path = tmp_path / "build"
+    subprocess.run(
+        [
+            "python",
+            str(BUILD_DOCS_PATH),
+            "develop",
+            str(build_path),
+            "--max-notebooks",
+            "0",
+            "--fail-on-warning",
+        ]
+    ).check_returncode()
```

Comment:
Might be nice to make a slightly nicer error message here to help a user understand what's gone wrong, currently it just fails on a `subprocess.CalledProcessError`, which might be a bit confusing. Judging by how broken are docs have been in the past, I guess it might happen a fair amount.

Reply:
> Might be nice to make a slightly nicer error message here to help a user understand what's gone wrong, currently it just fails on a `subprocess.CalledProcessError`, which might be a bit confusing. Judging by how broken are docs have been in the past, I guess it might happen a fair amount.

Sure, (though the point of this PR is to prevent us from getting into a badly broken state to begin with...)